### PR TITLE
Docker image improvements

### PR DESCRIPTION
**Pull request by dhuangnm**
1. Allowed generating docker image for both nightly and release
2. Removed build part, only installed wheels directly from nm pypi
3. Updated workflow files to automate the docker image generation

### CODE REVIEW

**Review by andy-neuma — APPROVED** [review]
cool. thanks.

File: .github/workflows/publish-docker.yml [hunk 1]
```diff
@@ -9,18 +9,35 @@ on:
   #   branches:
   #     - main
   workflow_dispatch:
+    inputs:
```
[the remark this answers is not in the document]
Comment (dhuangnm) [reply .github/workflows/publish-docker.yml:5]: done, thanks

File: Dockerfile [hunk 2]
```diff
@@ -31,67 +31,18 @@ COPY requirements-dev.txt requirements-dev.txt
 RUN --mount=type=cache,target=/root/.cache/pip \
```
[the remark this answers is not in the document]
Comment (dhuangnm) [reply Dockerfile:31]: thanks

File: .github/actions/nm-setup-nvidia-container-toolkit/action.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: set up nvidia-container-toolkit for docker
```
[the remark this answers is not in the document]
Comment (dhuangnm) [reply .github/actions/nm-setup-nvidia-container-toolkit/action.yml:1]: :)

**Pull request merged**